File: p2/p.py
#!/usr/bin/py

import sys

def fib():
    a = 1
    b = 2
    total = 0
    max = int(sys.argv[1])
    while a < max:
        if (a%2 == 0):
            total += a
        tmp = a + b
        a = b
        b = tmp
    return total
print (fib())

File: p2/ruby/p.rb
#!/usr/bin/ruby

def fib()
	a = 1; b = 2; tmp = 0, sum = 0
	((a%2 == 0) ? (sum+=a) : (); (tmp = a + b; a = b; b = tmp)) while a < Integer(ARGV[0])
	sum
end

puts(fib())

